kernel/reduction.ml

```diff
@@ -408,8 +409,12 @@ and eqappr cv_pb l2r infos (lft1,st1) (lft2,st2) cuniv =
     (* 2 constants, 2 local defined vars or 2 defined rels *)
     | (FFlex fl1, FFlex fl2) ->
       (try
-         let cuniv = conv_table_key infos.cnv_inf fl1 fl2 cuniv in
-         convert_stacks l2r infos lft1 lft2 v1 v2 cuniv
+         let nargs = CClosure.stack_args_size v1 in
```

Comment:
Can you introduce a function `arg_size : fstack -> fstack -> int` that returns the number of arguments if they coincide and raise `Not_convertible` otherwise? Then you can factor it out with the `FInd / FInd` case.